cmd/init.go

```diff
@@ -43,6 +44,124 @@ func (h *Handler) initNew(ctx context.Context, req *entity.CommandRequest) error
 	return nil
 }
 
+func (h *Handler) initFromTemplate(ctx context.Context, req *entity.CommandRequest) error {
+	// Prompt for a template to use
+
+	template, err := ui.PromptStarterTemplates()
+	if err != nil {
+		return err
+	}
+
+	// Select GitHub owner
+
+	ui.StartSpinner(&ui.SpinnerCfg{
+		Message: "Fetching GitHub scopes",
+	})
+	scopes, err := h.ctrl.GetWritableGithubScopes(ctx)
+	if err != nil {
+		return err
+	}
+	if len(scopes) == 0 {
+		return CLIErrors.NoGitHubScopesFound
+	}
+	ui.StopSpinner("")
+
+	owner, err := ui.PromptGitHubScopes(scopes)
+	if err != nil {
+		return err
+	}
+
+	// Enter project name
+
+	name, err := ui.PromptProjectName()
+	if err != nil {
+		return err
+	}
+
+	isPrivate, err := ui.PromptIsRepoPrivate()
+	if err != nil {
+		return err
+	}
+
+	// Prompt for env vars (if required)
+
+	variables := make(map[string]string)
+	if len(template.EnvVars) > 0 {
+		fmt.Printf("\n%s\n", ui.Bold("Environment Variables"))
+		for _, envVar := range template.EnvVars {
+			fmt.Printf("\n(%s)\n", ui.GrayText(envVar.Desc))
+			v, err := ui.PromptText(envVar.Name)
+			if err != nil {
+				return err
+			}
+			variables[name] = v
+		}
+		fmt.Print("\n")
+	}
```

Comment:
Environment variables can be optional and/or have a default value. This should probably handle that as well.
For optional env vars I think we could show a "(optional. enter to leave blank)" text (or something similar). For env vars with a default value, maybe we can auto-fill their input?

Reply:
Oo, will add that, nice catch